### PR TITLE
Added: Modal Component

### PR DESCRIPTION
**Pull request by hussiantabha**
Modal component with functionality

### CODE REVIEW

File: components/modal.css [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  left: 60%;
+  top: 60%;
+  z-index: 1;
+  background-color: var(--light-gray);
+  transform: translate(-50%, -50%);
+  display: flex;
+  flex-flow: column wrap;
+  justify-content: center;
+  align-items: center;
+  padding: 1rem 0;
```
Comment (Prabhav-Sharma) [inline components/modal.css:20]: Use global variables for spacing for properties like margin, padding, and gap to maintain consistency across the website

Comment (hussiantabha) [reply components/modal.css:20]: Will do this

File: docs.html [hunk 2]
```diff
@@ -725,6 +726,32 @@ <h1 class="display-title">Navbar</h1>
             </iframe>
           </div>
         </section>
+        <section class="display-code-content" id="modal">
+          <h1 class="display-title">Modal</h1>
+          <button class="btn btn-primary btn-modal">Show Modal</button>
+          <div class="modal-container">
+            <h1 class="modal-title">Modal Title</h1>
```
Comment (junaidshaikh-js) [inline docs.html]: There should be only one `h1` per page, read about heading hierarchy

**Review by junaidshaikh-js — COMMENTED** [review]
Good work!

File: docs.html (hunk 2 above)
Comment (hussiantabha) [reply docs.html]: Oh Yes will change it to h3 or h4 and then merge is it ok?

**Pull request merged**